Modules/EventHub/2.0/deploy.json

```diff
@@ -1,140 +1,324 @@
+    "tagProject": {
       "type": "string",
+      "defaultValue": "",
       "metadata": {
-        "description": "Optional. Authorization Rules Root Managed Shared Access Key."
+        "description": "Optional. The name of the project"
+      }
+    },
+    "tagApplication": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The name of the application"
+      }
+    },
+    "tagOwner": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The business owner for the application"
+      }
+    },
+    "tagOwnerEmail": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Optional. The Email address of the business owner for the application"
       }
     }
   },
   "variables": {
+    "namespaceName": "[parameters('namespaceName')]",
+    "namespaceResourceId": "[resourceId('Microsoft.EventHub/Namespaces',parameters('namespaceName'))]",
+    "diagnosticStorageAccountId": "[parameters('diagnosticStorageAccountId')]",
+    "diagnosticWorkspaceId": "[parameters('logAnalyticsWorkspaceId')]",
+    "defaultSASKeyName": "RootManageSharedAccessKey",
+    "authRuleResourceId": "[resourceId('Microsoft.EventHub/namespaces/authorizationRules', parameters('namespaceName'), variables('defaultSASKeyName'))]",
+    "apiVersion": "[providers('Microsoft.EventHub','namespaces').apiVersions[0]]",
+    "pidName": "[concat('pid-', parameters('cuaId'))]"
   },
   "resources": [
     {
-      "apiVersion": "2017-04-01",
-      "name": "[parameters('namespaceName')]",
+      "condition": "[not(empty(parameters('cuaId')))]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2018-02-01",
+      "name": "[variables('pidName')]",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": []
+        }
+      }
+    },
+    {
       "type": "Microsoft.EventHub/namespaces",
+      "apiVersion": "[variables('apiVersion')]",
+      "name": "[parameters('namespaceName')]",
       "location": "[parameters('location')]",
+      "tags": {
+        "Environment": "[parameters('tagEnvironment')]",
+        "Application": "[parameters('tagApplication')]",
+        "Project": "[parameters('tagProject')]",
+        "Owner": "[parameters('tagOwner')]",
+        "OwnerEmail": "[parameters('tagOwnerEmail')]"
+      },
       "sku": {
-        "name": "[parameters('eventHubSku')]",
-        "tier": "[parameters('eventHubSku')]",
-        "capacity": "[parameters('skuCapacity')]"
+        "name": "Standard",
+        "tier": "Standard",
+        "capacity": 1
       },
-      "tags": {
-        "displayName": "[parameters('namespaceName')]"
+      "properties": {
+        "zoneRedundant": false,
```

Comment:
I feel like these values must come as input parameters and we can set defaults if we'd like, try to limit the use of fixed values.